sdk/src/logs/logger.cc

```diff
@@ -36,23 +51,76 @@ void Logger::log(const opentelemetry::logs::LogRecord &record) noexcept
 
   // TODO: Sampler logic (should include check for minSeverity)
 
-  /**
-   * Convert the LogRecord to the heap first before sending to processor.
-   * TODO: Change the API log(LogRecord) function to log(*LogRecord) so the following line
-   * converting record a heap variable can be removed
-   */
-  auto record_pointer =
```

Comment:
So this code was throwing `std::bad_alloc` before your change, right?

Reply:
Not that I'm aware of, but updated this to use a recordable for faster/more reliable perfomance  (#403), and also allow custom recordable implementations by exporters